JShellAPI/src/test/java/org/togetherjava/jshellapi/JShellApiTests.java

```diff
@@ -0,0 +1,90 @@
+package org.togetherjava.jshellapi;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import org.togetherjava.jshellapi.dto.JShellResult;
+import org.togetherjava.jshellapi.dto.JShellSnippetResult;
+import org.togetherjava.jshellapi.dto.SnippetStatus;
+import org.togetherjava.jshellapi.dto.SnippetType;
+import org.togetherjava.jshellapi.rest.ApiEndpoints;
+
+import java.time.Duration;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This class holds integration tests for JShellAPI. It depends on gradle building image task, fore
+ * more information check "test" section in gradle.build file.
+ *
+ * @author Firas Regaieg
+ */
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class JShellApiTests {
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @Test
+    @DisplayName("When posting code snippet, evaluate it then returns successfully result")
+    public void evaluateCodeSnippetTest() {
+
+        final String testEvalId = "test";
+
+        // -- performing a first code snippet execution
+
+        final String firstCodeExpression = "int a = 2+2;";
+
+        final JShellSnippetResult firstCodeSnippet = new JShellSnippetResult(SnippetStatus.VALID,
+                SnippetType.ADDITION, 1, firstCodeExpression, "4");
+        final JShellResult firstCodeExpectedResult =
+                getJShellResultDefaultInstance(firstCodeSnippet);
+
+        assertThat(testEval(testEvalId, firstCodeExpression)).isEqualTo(firstCodeExpectedResult);
+
+        // -- performing a second code snippet execution
+
+        final String secondCodeExpression = "a * 2";
+
+        final JShellSnippetResult secondCodeSnippet = new JShellSnippetResult(SnippetStatus.VALID,
+                SnippetType.ADDITION, 2, secondCodeExpression, "8");
+
+        final JShellResult secondCodeExpectedResult =
+                getJShellResultDefaultInstance(secondCodeSnippet);
+
+        assertThat(testEval(testEvalId, secondCodeExpression)).isEqualTo(secondCodeExpectedResult);
+    }
```

Comment:
Why do you skip every even line, you are wasting space for no reason.
This whole code could be done in 2 or 4 lines instead of 30, no more than that.
You add comments because your code is becoming unreadable because you made it take so much place.
You are creating variable when you don't need to, there is no reason to be that verbose.